route.go

```diff
@@ -9,14 +9,16 @@ type Route struct {
 	Parts     []string
 	Layout    *Fragment
 	fragments []*Fragment
+	Metadata  map[string]string
 }
 
-func newRoute(path string, layout *Fragment, fragments []*Fragment) *Route {
+func newRoute(path string, metadata map[string]string, layout *Fragment, fragments []*Fragment) *Route {
```

Comment:
```suggestion
func newRoute(path string, layout *Fragment, fragments []*Fragment, metadata map[string]string) *Route {
```

Super minor, but thoughts on putting the metadata last? That way it's less churn if we remove it or change the API later?